src/test/java/hudson/plugins/analysis/core/IsOverriddenTest.java

```diff
@@ -0,0 +1,73 @@
+package hudson.plugins.analysis.core;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import org.junit.Test;
+
+import hudson.Launcher;
+import hudson.matrix.MatrixAggregator;
+import hudson.matrix.MatrixBuild;
+
+import hudson.model.BuildListener;
+
+import hudson.plugins.analysis.core.HealthAwarePublisher;
+
+import static org.junit.Assert.*;
+
+/**
+ * Test for {@link HealthAwarePublisher.isOverridden} method.
+ */
+public class IsOverriddenTest {
+
+    /**
+     * Test that a method is found by isOverriden even when it is inherited from an intermediate class.
+     */
+    @Test
+    public void isOverridenTest() throws NoSuchMethodException, SecurityException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+        Method method = HealthAwarePublisher.class.getDeclaredMethod("isOverridden", Class.class, Class.class, String.class, Class[].class);
+        method.setAccessible(true);
```

Comment:
Uh, why not simply make the method default access (“package private”) so you can call it directly from the test?